src/lara-plugin/plugin-api/decorate-content.ts

```diff
@@ -1,8 +1,28 @@
 import * as TextDecorator from "@concord-consortium/text-decorator";
+import { observable } from "mobx";
+
+export interface ITextDecorationInfo {
+  words: string[];
+  replace: string;
+  wordClass: string;
+  eventListeners: IEventListeners;
+}
+
+export interface IPluginEvent {
+  type: string,
+  text: string,
+}
+
+export let textDecorationInfo: ITextDecorationInfo = observable({
+  words: [],
+  replace: "",
+  wordClass: "",
+  eventListeners: [],
+});
 
 export interface IEventListener {
   type: string;
-  listener: (evt: Event) => void;
+  listener: (evt: Event | IPluginEvent) => void;
```

Comment:
I'd recommend making this more explicitly typed. So it would be more like a `IDecoratedContentEventListener` and `IDecoratedContentEvent`.  
I don't think you need to keep the generic `Event` option since this code is living in the Activity Player it doesn't need to support sending these old style events.

The place where the backward compatibility is in the glossary plugin and the LARA plugin api types in the LARA codebase which the glossary plugin is using.

Reply:
Don't these need to sync up with these types?
https://github.com/concord-consortium/glossary-plugin/blob/bd3d9e1d2f2ab22b76c480ed29adaf58add4ea23/src/components/plugin/plugin-app.tsx#L302
And isn't the existing implementation of `IEventListener ` used below in `decorateContent`?

I don't understand what we're trying to accomplish here.